documentation/authentication.md

```diff
@@ -0,0 +1,139 @@
+If you're running in Kubernetes, we recommend creating secrets and mounting them into container via a deployment manifest. You can use this `kubectl` command to either create or rotate the API Key.
+
+```sh
+kubectl create secret generic apikey \
+	--from-literal=ApiAuthentication__ApiKeyHash=$hash \
+	--dry-run=client -o yaml \
+	| kubectl apply -f -
+```
+
+Mount the secret as a file into `/etc/dotnet-monitor` in the deployment manifest for your application (abbreviated for clarity). `dotnet-monitor` only supports automatic key rotations for secrets mounted as volumes. For other kinds of secrets, the `dotnet-monitor` process must be restarted for the new key to take effect. 
+
+```yaml
+spec:
+  volumes:
+  - name: apikey
+    secret:
+      secretName: apikey
+  containers:
+  - name: dotnetmonitoragent
+    image: mcr.microsoft.com/dotnet/dotnet-monitor:5.0.0-preview.4
+    volumeMounts:
+      - name: apikey
+        mountPath: /etc/dotnet-monitor
+
+```
+
+> **NOTE:** For a complete example of running dotnet-monitor in Kubernetes, see [Running in a Kubernetes Cluster](getting-started.md#running-in-a-kubernetes-cluster) in the Getting Started guide.
+
+### Generate an API Key with PowerShell
+```powershell
+$rng = New-Object System.Security.Cryptography.RNGCryptoServiceProvider
+$secret = [byte[]]::new(32)
+$rng.GetBytes($secret)
+$API_KEY = [Convert]::ToBase64String($secret)
+"Authorization: MonitorApiKey $API_KEY"
+
+$secretStream = [System.IO.MemoryStream]::new($secret)
+$HASHED_KEY = (Get-FileHash -Algorithm SHA256 -InputStream $secretStream).Hash
+$rng.Dispose()
+
+Write-Output "ApiKeyHash: $HASHED_KEY"
+Write-Output "ApiKeyHashType: SHA256"
+```
+
+### Generate an API Key with a Linux shell
+
+```sh
+API_KEY=`openssl rand -base64 32`
+echo "Authorization: MonitorApiKey $API_KEY"
+
+HASHED_KEY=`$API_KEY | base64 -d | sha256sum`
+echo "ApiKeyHash: $HASHED_KEY"
+echo "ApiKeyHashType: SHA256"
+```
+
+## Authenticating requests
+When using Windows Authentication, your browser will automatically handle the Windows authentication challenge. If you are using an API Key, you must specify it via the authentication header.
+
+```sh
+curl -H "Authorization: MonitorApiKey Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=" https://localhost:52323/processes
+```
+
+If using PowerShell, be sure to use `curl.exe`, as `curl` is an alias for `Invoke-WebRequest` that does not accept the same parameters.
+
+```powershell
+curl.exe -H "Authorization: MonitorApiKey Uf6Yq8ZGkcn+ltq2QLHxuXpKA6/Q5VH5Mb5aSIJBxRc=" https://localhost:52323/processes
+```
+
+
+
+## Disabling Authentication
+
+Disabling authentication could enable lower privileged processes to exfiltrate sensitive information, such as the full contents of memory, from any .NET Core application running within the same boundary. You should only disable authentication when you have evaluated and mitigated the security implications of running `dotnet-monitor` unauthenticated.
```

Comment:
```suggestion
Disabling authentication could enable lower privileged processes to exfiltrate sensitive information, such as the full contents of memory, from any .NET application running within the same boundary. You should only disable authentication when you have evaluated and mitigated the security implications of running `dotnet-monitor` unauthenticated.
```

nit: Drop Core to avoid any .NET 5/6 confusion